ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java

```diff
@@ -115,17 +117,28 @@ private PlanNode buildQueryLogicalPlan(
     if (logicalPlan instanceof KsqlStructuredDataOutputNode) {
       KsqlStructuredDataOutputNode ksqlStructuredDataOutputNode =
           (KsqlStructuredDataOutputNode) logicalPlan;
-
-      StructuredDataSource
+      final StructuredDataSource
           structuredDataSource =
-          new KsqlStream(
-              sqlExpression,
-              ksqlStructuredDataOutputNode.getId().toString(),
-              ksqlStructuredDataOutputNode.getSchema(),
-              ksqlStructuredDataOutputNode.getKeyField(),
-              ksqlStructuredDataOutputNode.getTimestampExtractionPolicy(),
-              ksqlStructuredDataOutputNode.getKsqlTopic()
-          );
+          (ksqlStructuredDataOutputNode.getNodeOutputType() == DataSourceType.KTABLE)
```

Comment:
For my own understanding, how is this working exactly? The `structuredDataSource` object seems to only be used if `analysis.isDoCreateInto` is `true`, which only seems to be the case for `INSERT INTO` statements. This is the only place where `getNodeOutputType` is called. So how does this change the behavior for non `INSERT INTO` statements?

Reply:
The opposite, indeed `isDoCreateInto` is true only for CSAS/CTAS and it's false for `INSERT INTO`.  For `INSERT INTO` we already have the output stream and no need to set it's type!

Reply:
ah. thanks for the clarification.